### DISCUSSION TITLE
Community container Latest/Beta tag

### DISCUSSION THREAD

**Discussion by docjyJ** [nextcloud/all-in-one#4784] [Ideas] [answered]
### Is your feature request related to a problem? Please describe. 

The AIO containers have a Latest and Beta tag, for the community containers it would be good to have a Latest and Beta tag too, to have a way to test the functionality.

### Describe the solution you'd like

Add beta tag to community containers

### Describe alternatives you've considered

Idk

### Additional context

No

**Comment by szaimen** [comment, marked as answer]
Hi, interesting idea! However something comparable is already possible and rather the way I would suggest: 
if you want the community to test something for you, you can simply adjust the image_tag in the json, e.g. by increasing it one version and it will be tested first by people running the beta channel and afterwards it will be pushed to stable.
I hope this helps?

**Reply by docjyJ** [reply]
Yes, I understand. But Nextcloud AIO does not have the same maintenance cycle as the community.

Some features take time to be properly implanted, as for Stalwart.

The Nextcloud beta is often too short to guarantee stability in the update of the community container.

I hope I am understandable?

**Reply by szaimen** [reply]
I see the problem. What would be your suggestion? How could this be implemented in terms of logic?

**Reply by docjyJ** [reply]
```json
{
  "aio_services_v1": [
    {
      "_comment": "...",
      "image": "my_user/my_image",
      "image_tag": "my_tag",
      "image_beta_tag": "my_beta",
      "_comment": "..."
    }
  ]
}
```
When the AIO is installing with the tag “latest”, the “image_tag” is used.
For other unstable installation, it is  “image_beta_tag” which is used.
For compatibility, if “image_beta_tag” is not put, then “image_tag” is used by default.

**Reply by docjyJ** [reply]
It looks easy to implement.
If you think it's a good idea, I can try to make a PR.

**Reply by Zoey2936** [reply]
but what if the beta version would require diffrent mounts/env/ports etc. this would mean that either beta would break or latest

**Reply by docjyJ** [reply]
> but what if the beta version would require diffrent mounts/env/ports etc.

It is more complicated, for this specific case we can keep the same method.

PR -> Beta -> Release

Probably a change of port, environment, or volume involves user intervention for migration.
So, no perfect solution...

**Reply by szaimen** [reply]
> but what if the beta version would require diffrent mounts/env/ports etc. this would mean that either beta would break or latest

Yes, true, this is the main reason why it was implemented like how it is currently...

**Reply by docjyJ** [reply]
Yes, I think the majority of updates do not contain breaking changes.

If the addition of a beta tag is simple, it is an addition that improves the stability of the community container.
